p2p/src/peers/mod.rs

```diff
@@ -1,85 +1,271 @@
 //! Library for managing a list of available peers
 
-use serde::{Deserialize, Serialize};
-
-use std::collections::HashMap;
-use std::net::SocketAddr;
-
 use rand::{thread_rng, Rng};
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, net::SocketAddr, u8};
 
+use witnet_crypto::hash::calculate_sha256;
 use witnet_util::timestamp::get_timestamp;
 
 /// Peer information being used while listing available Witnet peers
 #[derive(Serialize, Deserialize)]
 struct PeerInfo {
     address: SocketAddr,
-    _timestamp: i64,
+    timestamp: i64,
 }
 
 /// Peers TBD
 #[derive(Default, Serialize, Deserialize)]
 pub struct Peers {
-    /// Server sessions
-    peers: HashMap<SocketAddr, PeerInfo>,
+    /// Bucket for tried addresses
+    tried_bucket: HashMap<u16, PeerInfo>,
+    /// Bucket for new addresses
+    new_bucket: HashMap<u16, PeerInfo>,
+    /// Nonce value
+    sk: u8,
```

Comment:
@girazoki something to say about this `u8`?

Reply:
@girazoki say that better use a u64